src/access-control.guard.ts

```diff
@@ -11,16 +11,26 @@ export class ACGuard implements CanActivate {
     @InjectRolesBuilder() private readonly roleBuilder: RolesBuilder,
   ) {}
 
-  canActivate(context: ExecutionContext): boolean {
+  async getUser(context: ExecutionContext): Promise<any> {
```

Comment:
Maybe it would be nice to make that method more general over T with fallback to `any`
`getUser<T = any>(...): Promise<T> { ... }`

Reply:
Overall, Thank you for your Contribution. :smile:

Reply:
I've updated this and fixed some other typings.

However I didn't use `getUser<T = any>` to do it. Partially because that's really just a type assertion (i.e. `as`) [disguised as a generic](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes). But also since it doesn't actually typecheck anything, getUser won't be restricted because inside the definition it always gets the default `any` and `getUserRoles` also doesn't even pass a type.

Instead I used a class generic to do it. However `<T = any>` did not work, typescript kept saying `.roles` was not defined on `T = any`. So I went with `<T extends any = any>`.